Server/src/main/java/org/gluu/oxauth/model/authorize/AuthorizeParamsValidator.java

```diff
@@ -35,10 +35,12 @@ public static boolean validateParams(List<ResponseType> responseTypes, List<Prom
             return false;
         }
 
-        if (responseTypes.contains(ResponseType.TOKEN) || responseTypes.contains(ResponseType.ID_TOKEN)) {
-            if (StringUtils.isBlank(nonce)) {
-                return false;
-            }
+        boolean existsNonce = StringUtils.isNotBlank(nonce);
+        if (!existsNonce && ((responseTypes.contains(ResponseType.CODE) && responseTypes.contains(ResponseType.ID_TOKEN))
+                || (responseTypes.contains(ResponseType.ID_TOKEN) && responseTypes.size() == 1)
+                || (responseTypes.contains(ResponseType.ID_TOKEN) && responseTypes.contains(ResponseType.TOKEN))
+                || (responseTypes.contains(ResponseType.TOKEN) && responseTypes.size() == 1))) {
+            return false;
```

Comment:
We should retire this `validateParams`, add new `validateResponseType` and throw error directly
```
public void validateResponseType(List<ResponseType> responseTypes, String nonce) {
     if (!existsNonce && ((responseTypes.contains(ResponseType.CODE) && responseTypes.contains(ResponseType.ID_TOKEN))
                || (responseTypes.contains(ResponseType.ID_TOKEN) && responseTypes.size() == 1)
                || (responseTypes.contains(ResponseType.ID_TOKEN) && responseTypes.contains(ResponseType.TOKEN))
                || (responseTypes.contains(ResponseType.TOKEN) && responseTypes.size() == 1))) {
        throw new WebApplicationException(...);
}
```